### PR TITLE
Fixes #32704 - Improve parent scope identification

### PR DESCRIPTION
**Pull request by tbrisker**
When a parent resource has a name that appears like it is
parameterized but actually isn't (such as `1-name`), we try to find
the parent scope from the resource name but fail (since the actual id of
the record is different). Instead of returning an empty scope, fall back
to friendly_find which may find the resource correctly.

### CODE REVIEW

File: app/controllers/concerns/application_shared.rb [hunk 1]
```diff
@@ -106,9 +106,10 @@ def scope_by_resource_id(base_scope, resource_id)
     return base_scope.none if resource_id.blank?
 
     if resource_id.is_a? String
-      # The class is parameterizable and the id is in '123-myname' format, extract the id from it
-      if base_scope.klass.respond_to?(:to_param) && resource_id.start_with?(/\d+-/)
```
Comment (tbrisker) [inline app/controllers/concerns/application_shared.rb:110]: ActiveRecord::Base defines `to_param` (and we add our own implementation in https://github.com/theforeman/foreman/blob/b9c42d48ba4515c0ff1160d4d551ee5a0d7702c2/app/models/concerns/parameterizable.rb) so this check is not needed

**Comment by ezr-ondrej** [comment]
I wonder what would happen if the resource with such id exists, but it's completely different resource?

**Comment by tbrisker** [comment]
> I wonder what would happen if the resource with such id exists, but it's completely different resource?

That is a good question indeed... Let me try changing the implementation around a bit

**Review by ezr-ondrej — APPROVED** [review]
Works as expected, thanks @tbrisker! :+1:
I'd love a test, but it is not as crucial here.

**Pull request merged**